ste/mgr-JobMgr.go

```diff
@@ -703,7 +701,23 @@ func (jm *jobMgr) DeferredCleanupJobMgr() {
 	// Sleep for sometime so that all go routine done with cleanUp and log the progress in job log.
 	time.Sleep(60 * time.Second)
 
-	jm.CloseLog()
+	// Release the allocated memory
+	jm.freeJobMgrMemory()
+
+	jm.logger.CloseLog()
+}
+
+// freeJobMgrMemory frees the memory by setting nil to addresses, so that gc reclaim it in next cycle.
+func (jm *jobMgr) freeJobMgrMemory() {
+	jm.chunkStatusLogger = nil
+	jm.xferChannels.normalTransferCh = nil
+	jm.xferChannels.lowTransferCh = nil
+	jm.xferChannels.lowChunkCh = nil
+	jm.xferChannels.normalChunckCh = nil
+	jm.coordinatorChannels.lowTransferCh = nil
+	jm.coordinatorChannels.normalTransferCh = nil
+	jm.coordinatorChannels.partsChannel = nil
+	jm.xferChannels.partsChannel = nil
 }
```

Comment:
@nitin-deamon, I think instead of doing this (which hides the problem) we should use this as a way to root-cause the problem. Since the above fixes the leak this means the ref being held is in the form of some message in one of these channels. We can print the len() of each channel and find out which one of them is not empty and then fix it the right way, by draining all the queued messages from that channel.

Reply:
Is the following the only effective one?
jm.chunkStatusLogger = nil

Reply:
let's remove this dead code.